File: java/src/main/java/com/mlt/vector/VariableSizeVector.java
package com.mlt.vector;

import java.nio.ByteBuffer;
import java.nio.IntBuffer;

public abstract class VariableSizeVector<K> extends Vector<ByteBuffer, K> {
  protected IntBuffer lengthBuffer;

  public VariableSizeVector(String name, IntBuffer lengthBuffer, ByteBuffer dataBuffer, int size) {
    super(name, dataBuffer, size);
    this.lengthBuffer = lengthBuffer;
  }

  public VariableSizeVector(
      String name, BitVector nullabilityBuffer, IntBuffer lengthBuffer, ByteBuffer dataBuffer) {
    super(name, nullabilityBuffer, dataBuffer);
    this.lengthBuffer = lengthBuffer;
  }
}

File: ts/src/vector/variableSizeVector.ts
import type BitVector from "./flat/bitVector";
import Vector from "./vector";

export abstract class VariableSizeVector<T extends ArrayBufferView, K> extends Vector<T, K> {
    protected constructor(
        name: string,
        protected offsetBuffer: Uint32Array,
        dataBuffer: T,
        sizeOrNullabilityBuffer: number | BitVector,
    ) {
        super(name, dataBuffer, sizeOrNullabilityBuffer);
    }
}
